### PR TITLE
Fix tensor read (device->host path)

### PR DESCRIPTION
**Pull request by svuckovicTT**
Currently, we're calling to_memory_config to get tensor from device to host. This isn't correct, so we're changing to the approach used in ttnn lib, which is to call `FromDevice` + `ToLayout` ops, as can be seen in the `to_torch` method in `third_party/tt-metal/src/tt-metal/ttnn/ttnn/operations/core.py`:
```py
    if ttnn.is_tensor_storage_on_device(tensor):
        tensor = ttnn.from_device(tensor, cq_id=cq_id)

    if tensor.layout != ttnn.ROW_MAJOR_LAYOUT:
        tensor = tensor.to(ttnn.ROW_MAJOR_LAYOUT, device)
```

This PR updates the conversion of `ToLayout` op from TTIR to TTNN. It also adds the `FromDevice` op.

### CODE REVIEW

**Comment by jnie-TT** [comment]
@svuckovicTT assuming you're getting this from core.py, I think there's a fix going in that flips these two operations: essentially untilize on device first before moving to host.

**Comment by svuckovicTT** [comment]
> @svuckovicTT assuming you're getting this from core.py, I think there's a fix going in that flips these two operations: essentially untilize on device first before moving to host.

Oh awesome, thanks for the heads up! I'll test it and update accordingly. Do you happen to have a link to their issue/PR?

**Comment by jnie-TT** [comment]
> > @svuckovicTT assuming you're getting this from core.py, I think there's a fix going in that flips these two operations: essentially untilize on device first before moving to host.
> 
> Oh awesome, thanks for the heads up! I'll test it and update accordingly. Do you happen to have a link to their issue/PR?

This is the PR AFAIK: https://github.com/tenstorrent/tt-metal/pull/12001

**Review by rpavlovicTT — APPROVED** [review]
LGTM after reordering calls as Jackson pointed out

**Review by sdjordjevicTT — APPROVED** [review]

File: lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp
Comment (nsmithtt) [inline lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp:198]: If the operand is already row major maybe we don't need to emit this layout op that would otherwise be a nop?

Comment (rpavlovicTT) [reply lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp:198]: @svuckovicTT is on PTO right now, and assigned me to check in this change after it's approved. Is it okay to leave it for him to fix toLayout no-op when he gets back? I can create a follow up issue.

Comment (svuckovicTT) [reply lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp:198]: Chiming in from the airport - cheers to delayed flights! :)

Yeah, we could avoid it, I'll create a follow-up task (can't VPN from Hungary, so can't test the runtime...). However, I think it's okay to check this in as-is for now, since it'll end up being a nop - the method checks for this and returns if the layout is already set to the desired one.

Comment (svuckovicTT) [reply lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp:198]: https://github.com/tenstorrent/tt-mlir/issues/773

**Review by nsmithtt — APPROVED** [review]

**Pull request merged**
